editor.planx.uk/src/@planx/components/Section/model.ts

```diff
@@ -10,3 +13,71 @@ export const parseSection = (
   title: data?.title || "",
   ...parseMoreInformation(data),
 });
+
+export function computeSectionStatuses({
+  sectionNodes,
+  currentCard,
+  breadcrumbs,
+  cachedBreadcrumbs,
+  isReconciliation,
+  alteredSectionIds,
+}: {
+  sectionNodes: Record<string, SectionNode>;
+  currentCard: Store.node | null;
+  breadcrumbs: Store.breadcrumbs;
+  cachedBreadcrumbs?: Store.cachedBreadcrumbs;
+  isReconciliation?: boolean;
+  alteredSectionIds?: string[];
+}): Record<string, SectionStatus> {
+  const seenPastNodeIds = Object.keys({ ...breadcrumbs });
+  const seenUpcomingNodeIds = cachedBreadcrumbs
+    ? Object.keys({ ...cachedBreadcrumbs })
+    : [];
+  const allSeenNodeIds = [...seenPastNodeIds, ...seenUpcomingNodeIds];
+  const currentCardId = currentCard?.id;
+  const sectionNodeIds = Object.keys(sectionNodes);
+
+  let hasPastCurrentSection = false;
+  let sectionStatuses: Record<string, SectionStatus> = {};
+  for (const [index, sectionId] of Object.entries(sectionNodeIds)) {
+    // check for an altered sections
+    if (alteredSectionIds && alteredSectionIds!.includes(sectionId)) {
+      sectionStatuses[sectionId] = SectionStatus.NeedsUpdated;
+      hasPastCurrentSection = true;
+      continue;
+    }
+    // check for seen/current sections
+    if (!hasPastCurrentSection) {
+      const nextSectionId = sectionNodeIds.at(Number(index) + 1);
+      const seenNextSection =
+        nextSectionId && allSeenNodeIds.includes(nextSectionId);
+      if (
+        !allSeenNodeIds.includes(sectionId) ||
+        (isReconciliation && !seenNextSection && currentCardId === sectionId)
+      ) {
+        sectionStatuses[sectionId] = SectionStatus.ReadyToStart;
+        hasPastCurrentSection = true;
+      } else if (
+        seenUpcomingNodeIds.includes(sectionId) ||
+        (isReconciliation &&
+          !seenNextSection &&
+          currentCardId !== sectionId &&
+          !sectionNodeIds.includes(currentCardId!))
+      ) {
+        sectionStatuses[sectionId] = SectionStatus.ReadyToContinue;
+        hasPastCurrentSection = true;
+      } else {
+        sectionStatuses[sectionId] = SectionStatus.Completed;
+      }
+      continue;
+    }
+    // check for unseen sections
+    if (allSeenNodeIds.includes(sectionId)) {
+      sectionStatuses[sectionId] = SectionStatus.Started;
+    } else {
+      sectionStatuses[sectionId] = SectionStatus.NotStarted;
+    }
+  }
+
+  return sectionStatuses;
+}
```

Comment:
This function should ideally live in planx-core where it could also be used to compute the section statuses of exported data.

Reply:
Do we have an existing case of an export data format that will need the section _status_ ? I've been assuming that for exported application data, we always assume it's a finished app so all sections have been completed and we only care about sending the section _names_ as a grouping mechanism for the rest of the flow nodes? I figure statuses are really only valuable to applicants, not validation officers. 

Therefore I'd picture this function in the navigation store and not planx-core longterm!

Reply:
Yes, fair point. It's not helpful for exported data but it could still be helpful to determine the status of an application for pre-processing steps (for example, checking that a flow is complete before locking a session for invite-to-pay) but this isn't an immediate concern so this can stay here for now.